influxql/call_iterator.go

```diff
@@ -362,18 +360,30 @@ func booleanLastReduce(prev, curr *BooleanPoint) (int64, bool, []interface{}) {
 func NewDistinctIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	switch input := input.(type) {
 	case FloatIterator:
-		return &floatReduceSliceIterator{input: newBufFloatIterator(input), opt: opt, fn: floatDistinctReduceSlice}, nil
+		createFn := func() (FloatPointSliceAggregator, FloatPointSliceEmitter) {
```

Comment:
I think the interfaces need to be different. I could see combining the Emitter interface and just grabbing  a single value from a list for non slice but the Aggregation interface needs to be different. How were you thinking of combining the Agg interfaces?